### PR TITLE
Distributed tracing

### PR DESCRIPTION
**Pull request by serras**
Blocked until https://github.com/mtth/tracing/pull/2 is merged

### CODE REVIEW

File: instrumentation/tracing/src/Mu/Instrumentation/Tracing.hs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+zipkinTracing :: (MonadIO m, MonadTrace m)
+              => MuTracing -> RpcInfo i -> m a -> m a
+zipkinTracing zpk NoRpcInfo h =
+  rootSpan (samplingPolicy zpk) (rootName zpk) h
+zipkinTracing zpk (RpcInfo _ _ _ (M.fromList -> hdrs) _) h =
+  case getB3 of
+    Nothing  -> rootSpan (samplingPolicy zpk) (rootName zpk) h
+    Just spn -> serverSpan spn h
+  where getB3 =   (b3FromHeaderValue =<< M.lookup "b3" hdrs)
```
Comment (cb372) [inline instrumentation/tracing/src/Mu/Instrumentation/Tracing.hs:79]: "The B3 portion of the header is so named for the original name of Zipkin: BigBrotherBird." 😲

**Comment by cb372** [comment]
LGTM

**Comment by serras** [comment]
I am hesitant to merge until a new version of `tracing` supporting `MonadBaseControl` comes out.

**Comment by serras** [comment]
Updated to refer to the [forked version of the `tracing` library](https://hackage.haskell.org/package/tracing-control-0.0.6)

**Pull request merged**